src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java

```diff
@@ -190,5 +187,22 @@ public void deleteAll() {
 
         return domain;
     }
+
+    @Override
+    public <T> T update(@NonNull T object) {
+        @SuppressWarnings("unchecked")
+        final GremlinEntityInformation information = new GremlinEntityInformation(object.getClass());
+        @SuppressWarnings("unchecked")
+        final Class<T> domainClass = (Class<T>) object.getClass();
+
+        final GremlinSource source = information.getGremlinSource();
+        final T found = this.findById(source.getId(), domainClass);
+
+        if (found == null) {
+            throw new GremlinUpdationException("cannot update the object doesn't exist");
+        }
+
+        return this.insert(object);
```

Comment:
Is there any way we can update directly without find first?

Reply:
Yes, we can do that without specific id value, but if the node not exist in graph. The sever may generate one for you. Which is out of our control. So if you try to find after update, you may got different id for one domain.